wsmaster/che-core-api-factory-gitlab/src/main/java/org/eclipse/che/api/factory/server/gitlab/GitlabOAuthTokenFetcher.java

```diff
@@ -164,8 +163,12 @@ public PersonalAccessToken fetchPersonalAccessToken(Subject cheSubject, String s
       // validating personal access token from secret. Since PAT API is accessible only in
       // latest GitLab version, we just perform check by accessing something from API.
       try {
-        gitlabApiClient.getUser(personalAccessToken.getToken());
-        return Optional.of(Boolean.TRUE);
+        GitlabUser user = gitlabApiClient.getUser(personalAccessToken.getToken());
```

Comment:
Is it worth abstracting an interface from `gitlabApiClient` and `githubApiClient`` so that this check for PAT vaildity could be implemented just once and not repeated in fetchers for both github and gitlab in the same way?

Reply:
I agree it deserves some refactorig already, but it wouldn't be just as easy as exctract method due to different scope sets etc.
So i would better do it seprately.